metadata-ingestion/src/datahub/ingestion/source/delta_lake/source.py

```diff
@@ -126,24 +137,80 @@ def create(cls, config_dict: dict, ctx: PipelineContext) -> "Source":
         config = DeltaLakeSourceConfig.parse_obj(config_dict)
         return cls(config, ctx)
 
+    inner_data_type = ""
+    counter = 0
+
+    def _get_inner_field_details(self, raw_field):
+        if "elementType" in raw_field and type(raw_field.get("elementType")) == dict:
+            self.inner_data_type = (
+                self.inner_data_type
+                + "item:"
+                + str(raw_field.get("elementType").get("type"))
+                + "<"
+            )
+            self.counter = self.counter + 1
+            self._get_inner_field_details(raw_field.get("elementType"))
+        elif "fields" in raw_field and type(raw_field.get("fields")) == list:
+            data_type = ""
+            data_type = data_type + ",".join(
+                "{0}:{1}".format(field.get("name"), field.get("type"))
+                for field in raw_field.get("fields")
+            )
+            self.inner_data_type = self.inner_data_type + data_type
+        else:
+            self.inner_data_type = (
+                self.inner_data_type + "item:" + raw_field.get("elementType")
+            )
+
+    def _parse_datatype(self, raw_field_json_str):
+        raw_field_json = json.loads(raw_field_json_str)
+
+        # get the parent field name and type
+        field_name, field_type = _get_parent_field_details(raw_field_json)
+
+        # recursively parse the fields to get the native data type
+        self._get_inner_field_details(raw_field_json.get("type"))
```

Comment:
Since this is not returning anything, the name `get*` seems a but misleading to me.

Reply:
will update to `_inner_field_details`